### ISSUE TITLE
Filters/serde not being mapped to db

### ISSUE THREAD

**Issue by brudil** [joist-orm/joist-orm#1048]
I assume this is some weird edge case because we're using custom serde a lot and the custom part of it is abstracted out afaik in the orm itself.

Given a CustomSerde that looks like this

```ts
export const CountrySerde: CustomSerde<CountryValue, string> = {
  fromDb(value) {
    return CountryValue.cca3OrFail(value);
  },
  toDb(value) {
    return value.cca3;
  },
};
```

And a query that looks like
```ts
const marketCommentary = await em.get.findOne(MarketCommentary, {
    country,
});
```

<img width="847" alt="image" src="https://github.com/joist-orm/joist-orm/assets/382352/3d9316cf-f35f-4b59-9eb0-6f69d244f4d1">

I'd had a cursorily glance over the QueryParser.ts to see if anything obvious stuck out, will debug a bit further tomorrow when I get the chance.

Could the fact it's a `findOne` and not a `find` be significantly changing the execution path?

**Comment by stephenh** [comment]
Hi @brudil ! Thanks for filing the issue; I happened to be working on Joist today anyway, so I'm pretty sure this is fixed in #1049 . Lmk!

**Comment by brudil** [comment]
Ouch! How did I not spot that - pretty sure I wrote that initially too! Apologies, thanks for the fix :)
